apps/meteor/app/channel-settings/server/methods/saveRoomSettings.ts

```diff
@@ -199,64 +217,65 @@ const settingSavers = {
 
 		if (room.teamId && room.teamMain) {
 			const type = value === 'c' ? TEAM_TYPE.PUBLIC : TEAM_TYPE.PRIVATE;
-			Team.update(user._id, room.teamId, { type, updateRoom: false });
+			await Team.update(user._id, room.teamId, { type, updateRoom: false });
 		}
 	},
-	streamingOptions({ value, rid }) {
+	async streamingOptions({ value, rid }: { value: IRoom['streamingOptions']; rid: IRoom['_id'] }) {
 		saveStreamingOptions(rid, value);
 	},
-	readOnly({ value, room, rid, user }) {
+	async readOnly({ value, room, rid, user }: { value: boolean; room: IRoom; rid: IRoom['_id']; user: IUser }) {
 		if (value !== room.ro) {
 			saveRoomReadOnly(rid, value, user);
 		}
 	},
-	reactWhenReadOnly({ value, room, rid, user }) {
+	async reactWhenReadOnly({ value, room, rid, user }: { value: boolean; room: IRoom; rid: IRoom['_id']; user: IUser }) {
 		if (value !== room.reactWhenReadOnly) {
 			saveReactWhenReadOnly(rid, value, user);
 		}
 	},
-	systemMessages({ value, room, rid, user }) {
+	async systemMessages({ value, room, rid }: { value: boolean; room: IRoom; rid: IRoom['_id']; user: IUser }) {
 		if (JSON.stringify(value) !== JSON.stringify(room.sysMes)) {
-			saveRoomSystemMessages(rid, value, user);
+			saveRoomSystemMessages(rid, value);
 		}
 	},
-	joinCode({ value, rid }) {
-		Rooms.setJoinCodeById(rid, String(value));
+	async joinCode({ value, rid }: { value: string; rid: IRoom['_id'] }) {
+		await Rooms.setJoinCodeById(rid, String(value));
```

Comment:
The function can now be removed from the old model as it is no longer used anywhere